orbit/pkg/update/flag_runner.go

```diff
@@ -24,103 +23,64 @@ import (
 // It uses an OrbitConfigFetcher (which may be the OrbitClient with additional middleware), along
 // with FlagUpdateOptions to connect to Fleet
 type FlagRunner struct {
-	configFetcher OrbitConfigFetcher
-	opt           FlagUpdateOptions
-	cancel        chan struct{}
+	queueOrbitRestart context.CancelFunc
+	opt               FlagUpdateOptions
 }
 
 // FlagUpdateOptions is options provided for the flag update runner
 type FlagUpdateOptions struct {
-	// CheckInterval is the interval to check for updates
-	CheckInterval time.Duration
 	// RootDir is the root directory for orbit state
 	RootDir string
 }
 
 // NewFlagRunner creates a new runner with provided options
 // The runner must be started with Execute
-func NewFlagRunner(configFetcher OrbitConfigFetcher, opt FlagUpdateOptions) *FlagRunner {
+func NewFlagReceiver(queueOrbitRestart context.CancelFunc, opt FlagUpdateOptions) *FlagRunner {
 	return &FlagRunner{
-		configFetcher: configFetcher,
-		opt:           opt,
-		cancel:        make(chan struct{}),
+		queueOrbitRestart: queueOrbitRestart,
+		opt:               opt,
 	}
 }
 
-// Execute starts the loop checking for updates
-func (r *FlagRunner) Execute() error {
-	log.Debug().Msg("starting flag updater")
-
-	ticker := time.NewTicker(r.opt.CheckInterval)
-	defer ticker.Stop()
-
-	for {
-		select {
-		case <-r.cancel:
-			return nil
-		case <-ticker.C:
-			log.Debug().Msg("calling flags update")
-			didUpdate, err := r.DoFlagsUpdate()
-			if err != nil {
-				logging.LogErrIfEnvNotSet(constant.SilenceEnrollLogErrorEnvVar, err, "flags updates failed")
-			}
-			if didUpdate {
-				log.Info().Msg("flags updated, exiting")
-				return nil
-			}
-			ticker.Reset(r.opt.CheckInterval)
-		}
-	}
-}
-
-// Interrupt is the oklog/run interrupt method that stops orbit when interrupt is received
-func (r *FlagRunner) Interrupt(err error) {
```

Comment:
one change in behavior is that this can't be cancelled from the outside anymore, is that right? IMO it's okay

Reply:
Yes, although previously the only time that happened was when the client was shutting down